pydm/tests/widgets/test_enum_combo_box.py

```diff
@@ -0,0 +1,330 @@
+
+
+@pytest.mark.parametrize("index", [
+    0,
+    1,
+    -1,
+    25,
+    -25,
+])
+def test_internal_combo_box_activated_int(qtbot, signals, index):
+    """
+    Test the the capability of the widget's activated slot in sending out a new enum string index value.
+
+    Expectations:
+    The value sent out from the "activated" slot reflects the correct new index value.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        pytest-qt window for widget test
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    index : int
+        The new enum string index value
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    # Connect the send_value_signal also to the conftest's receiveValue slot to intercept and verify that the correct
+    # new index value is sent out
+    pydm_enumcombobox.send_value_signal[int].connect(signals.receiveValue)
+    pydm_enumcombobox.activated[int].emit(index)
+
+    assert signals.value == index
+
+
+# --------------------
+# NEGATIVE TEST CASES
+# --------------------
+
+@pytest.mark.parametrize("enums, expected_error_message", [
+    (None, "Invalid enum value '{0}'. The value is expected to be a valid list of string values.".format(None)),
+    ((None, "abc"), "Invalid enum type '{0}'. The expected type is 'string'.".format(type(None))),
+    ((None, 123.456), "Invalid enum type '{0}'. The expected type is 'string'".format(type(None))),
+    ((None, None, None), "Invalid enum type '{0}'. The expected type is 'string'".format(type(None))),
+    ((123,),  "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
+    ((123.45,),  "Invalid enum type '{0}'. The expected type is 'string'".format(type(123.45))),
+    ((123, 456), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
+    ((123.456, None), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123.456))),
+    (("spam", 123, "eggs", "ham"), "Invalid enum type '{0}'. The expected type is 'string'".format(type(123))),
+])
+def test_set_items_neg(qtbot, caplog, enums, expected_error_message):
+    """
+    Test sending setting the widget with an undefined list of enum strings.
+
+    Expectations:
+    The correct error message is logged.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        pytest-qt window for widget test
+    caplog : fixture
+        To capture the log messages
+    enums : tuple
+        A list of strings as enum strings (choices) to populate to the widget.
+    """
+    pydm_enumcombobox = PyDMEnumComboBox()
+    qtbot.addWidget(pydm_enumcombobox)
+
+    pydm_enumcombobox.set_items(enums)
+
+    for record in caplog.records:
+        assert record.levelno == ERROR
+    assert expected_error_message in caplog.text
+
+
+@pytest.mark.parametrize("values, selected_index, expected", [
+    (("ON", "OFF"), 3, ""),
```

Comment:
I wouldn't use `parameterize` if you are only testing with one set of inputs.

Reply:
Good point. I had another test case there, but must have inadvertently removed it.